src/scripts/eastern_kingdoms/searing_gorge/searing_gorge.cpp

```diff
@@ -68,26 +97,66 @@ struct npc_dorius_stonetenderAI : public npc_escortAI
 
     void Aggro(Unit* pWho) override
     {
-        DoScriptText(urand(0, 1) ? SAY_DORIUS_AGGRO_1 : SAY_DORIUS_AGGRO_2, m_creature, pWho);
+        switch (urand(0, 3))
+        {
+            case 0:
+                DoScriptText(SAY_DORIUS_AGGRO_1, m_creature, pWho);
+                break;
+            case 1:
+                DoScriptText(SAY_DORIUS_AGGRO_2, m_creature, pWho);
+                break;
+            case 2:
+                DoScriptText(SAY_DORIUS_INJURED_DWARF, m_creature);
+                break;
+        }
     }
 
     void WaypointReached(uint32 uiPointId) override
     {
         switch (uiPointId)
         {
-            case 20:
-                // ToDo: research if there is any text here!
-                float fX, fY, fZ;
-                for (uint8 i = 0; i < MAX_STEELSHIFTERS; ++i)
-                {
-                    m_creature->GetNearPoint(m_creature, fX, fY, fZ, 0, 15.0f, i * M_PI_F / 2);
-                    m_creature->SummonCreature(NPC_DARK_IRON_STEELSHIFTER, fX, fY, fZ, 0, TEMPSUMMON_TIMED_OR_DEAD_DESPAWN, 60000);
+            case 10:
+                DoScriptText(SAY_DORIUS_CATCH_MY_BREATH, m_creature);
+                m_creature->SetStandState(UNIT_STAND_STATE_KNEEL);
+                break;
+            case 11:
+                SpawnSteelshifterWave(STEELSHIFTERS_COUNT_A);
+                break;
+            case 14:
+                if (Player* pPlayer = GetPlayerForEscort()) {
+                    m_creature->HandleEmote(EMOTE_ONESHOT_POINT);
+                    DoScriptText(SAY_DORIUS_THIS_IS_THE_PLACE, m_creature, pPlayer);
                 }
                 break;
-            case 33:
-                // ToDo: research if there is any event and text here!
+            case 15:
+                if (Player* pPlayer = GetPlayerForEscort())
+                    m_creature->SetFacingToObject(pPlayer);
+                DoScriptText(SAY_DORIUS_ONWARD, m_creature);
+                break;
+            case 19:
+                if (Player* pPlayer = GetPlayerForEscort())
+                    m_creature->SetFacingToObject(pPlayer);
+                DoScriptText(SAY_DORIUS_QUIT_SMOKING, m_creature);
+                m_creature->SetStandState(UNIT_STAND_STATE_KNEEL);
+                break;
+            case 20:
+                SpawnSteelshifterWave(STEELSHIFTERS_COUNT_B);
+                break;
+            case 23:
+                DoScriptText(SAY_DORIUS_RUN, m_creature);
+                break;
+            case 24:
+                // TODO: Start running from this point.
```

Comment:
m_creature->SetWalk(false);

Reply:
This doesn't seem to make a difference. I tried `m_creature->SetWalk(false, true);` as well, to see if `UNIT_STAT_RUNNING` would do anything, but no dice.

Reply:
Figured this out, need to use SetRun so the escortAI doesn't override it immediately.